src/state/transactions/hooks.tsx

```diff
@@ -89,6 +94,15 @@ export function useIsTransactionConfirmed(transactionHash?: string): boolean {
   return Boolean(transactions[transactionHash].receipt)
 }
 
+export function useSwapTransactionStatus(swapResult: SwapResult | undefined): TransactionStatus | undefined {
+  const { chainId } = useWeb3React()
+  const transaction = useTransaction(swapResult?.type === TradeFillType.Classic ? swapResult.response.hash : undefined)
+  const tokens = useAllTokensMultichain()
+
+  const swapActivity = transaction && chainId ? transactionToActivity(transaction, chainId, tokens) : undefined
+  return swapActivity?.status
```

Comment:
I think converting to the Activity type is overkill here. I suggest pulling the following  code from mini portfolio into a helper fn:

```
const status = !details.receipt
      ? TransactionStatus.Pending
      : details.receipt.status === 1 || details.receipt?.status === undefined
      ? TransactionStatus.Confirmed
      : TransactionStatus.Failed
```

so you can just do this:
```suggestion
  return getTransactionStatus(transaction)
```

Reply:
sure